JavaScript/QuickSort.js

```diff
@@ -0,0 +1,46 @@
+var items = [5,3,7,6,2,9];
+function swap(items, leftIndex, rightIndex){
+    var temp = items[leftIndex];
+    items[leftIndex] = items[rightIndex];
+    items[rightIndex] = temp;
+}
+function partition(items, left, right) {
+    var pivot   = items[Math.floor((right + left) / 2)], //middle element
+        i  = left,
+        j  = right; 
+    while (i <= j) {
+        while (items[i] < pivot) {
+            i++;
+        }
+        while (items[j] > pivot) {
+            j--;
+        }
+        if (i <= j) {
+            swap(items, i, j); //sawpping two elements
+            i++;
+            j--;
+        }
+    }
+    return i;
+}
+
+function quickSort(items, left, right) {
+    var index;
+    if (items.length > 1) {
+        index = partition(items, left, right);
+        if (left < index - 1) {
+            quickSort(items, left, index - 1);
+        }
+        if (index < right) { 
+            quickSort(items, index, right);
+        }
+    }
+    return items;
+}
+
+var sortedArray = quickSort(items, 0, items.length - 1);
+console.log(sortedArray); 
+
+//Input : 5 3 7 6 2 9
+//OutPut: 2 3 5 6 7 9
+//Time: O(nlogn)

```

Comment:
Can you please add space complexity and add link to your code in README.md please.

Reply:
And also use multiline comment to write sample I/O and time and space complexity.

Reply:
Okay I will do that